### ISSUE TITLE
How to adjust `_partition_bytes`?

### ISSUE THREAD

**Issue by hbsun2113** [bytedance/byteps#158]
Hi, I notice that `_partition_bytes` is bound to 4096000 by default and I want to know why. 
Besides, I want to adjust the `_partition_bytes`. And I also find that `_partition_bytes` has an impact on the number of partitions which should be less than `2^16`. I wonder if the size of tensor is too large and `_partition_bytes` is too small, whether the number of partitions will exceed  `2^16`?
![image](https://user-images.githubusercontent.com/27994442/69408402-bfcadd80-0d41-11ea-8fe8-779b8f9e5b47.png)

In other words, 4096000*(2^16)=268435456000, does it mean that the size of tensor should less than 268435456000?

**Comment by bobzhuyb** [comment]
BYTEPS_PARTITION_BYTES can set the partition size.

**Comment by hbsun2113** [comment]
> BYTEPS_PARTITION_BYTES can set the partition size.

Thank you for your response. I know how to set the partition size, but I do not know why you set it to 4096000, can you explain it?

**Comment by bobzhuyb** [comment]
Smaller partition size can give you better pipeline for overlapping PUSH and PULL. But small partition also means higher system implementation overhead. 4MB is a value based on our environment. You can tune it for the best performance in your environment.
